packages/cozy-realtime/src/Socket.js

```diff
@@ -0,0 +1,145 @@
+/* global WebSocket */
+import MicroEE from 'microee'
+import pickBy from 'lodash/pickBy'
+
+/**
+ * Socket class
+ *
+ * @class
+ */
+class Socket {
+  /**
+   * Doctype name on cozy stack
+   *
+   * @type {String}
+   */
+  _doctype = 'io.cozy.websocket'
+
+  /**
+   * An opened socket
+   *
+   * @type {WebSocket}
+   */
+  _webSocket = null
+
+  /**
+   * Function to return cozy client url
+   *
+   * @type {Function}
+   */
+  _url = null
+
+  /**
+   * Function to return cozy client token
+   *
+   * @type {Function}
+   */
+  _token = null
+
+  /**
+   * Handle WebSocket
+   *
+   * @see https://developer.mozilla.org/en-US/docs/Web/API/WebSockets_API/Writing_WebSocket_client_applications#Creating_a_WebSocket_object
+   *
+   * @constructor
+   * @param {Function} getUrl  Function to return cozy client url
+   * @param {Function} getToken  Function to return cozy client token
+   */
+  constructor(getUrl, getToken) {
+    this._getUrl = getUrl
+    this._getToken = getToken
+
+    this.removeAllListeners = this.removeAllListeners.bind(this)
```

Comment:
Nit: Maybe add a little comment to point out that `removeAllListeners` comes from `MicroEE`.

Reply:
j'ai résolu dans ce commit https://github.com/cozy/cozy-libs/pull/424/commits/f91c89862ccea84ab64ac7e0b54c459f5d035afd